packages/ag-charts-community/src/chart/legend.ts

```diff
@@ -988,6 +993,30 @@ export class Legend extends BaseProperties {
         // Fallback to returning closest match to the left/up.
         return closestLeftTop.datum;
     }
+    private getLegendItemHitbox(legendItemNode: { getBBox(): BBox }): BBox {
+        const bbox = legendItemNode.getBBox().clone();
+        bbox.grow(this.item.paddingX / 2, 'horizontal');
+        bbox.grow(this.item.paddingY / 2, 'vertical');
+        return bbox;
+    }
+    private getLegendItemHitboxProvider(legendItemNode: LegendMarkerLabel): BBoxProvider<BBox> {
+        // FIXME: This method is required for the `focusable` property of `createProxyElement`, but the `focusable` property should eventually be removed because it's only required to update the FocusIndicator
+        const legend = this;
+        return {
+            get id() {
+                return legendItemNode.id;
+            },
+            get visible() {
+                return legendItemNode.visible;
+            },
+            toCanvasBBox() {
+                return Transformable.toCanvas(legendItemNode.parent!, legend.getLegendItemHitbox(legendItemNode));
+            },
+            fromCanvasPoint(x: number, y: number) {
+                return Transformable.fromCanvasPoint(legendItemNode!, x, y);
+            },
+        };
```

Comment:
Nit: Maybe it would be better to add another sub-class of `NodeRegionBBoxProvider` to `regions.ts` which allows an alternative nodes coordinate space to be used, as `legend.ts` is already quite bloated?

Reply:
I don't think this nit matters. This code only exists to provide a `focusable` parameter to `ProxyInteractionManager` interface. This `focusable` parameter is used to update the `FocusIndicator` bounds. But we're going to phase out some of `FocusIndicator` in favour of the [:focus-visible](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible) pseudo-class which we're already using in financial charts interactive elements.